src/ott/tools/plot.py

```diff
@@ -245,3 +260,159 @@ def animate(
+    indices = jnp.unravel_index(idx, n_s)
+
+    # Setttings for plot
+    markers = "svopxdh"
+
+    for j in range(self._top_k):
+      points = [ot.x_s[i][indices[i][j], :] for i in range(self._k)]
+      points = [points[i] for i in ccworder(jnp.array(points))]
+      alphas = np.linspace(0.6, 0.2, self._top_k - self._n)
+      alpha = 0.6 if j < self._n else alphas[j - self._n]
+      points = ptc.Polygon(
+          points,
+          fill=True,
+          linewidth=2,
+          color=self._cmap[j > self._n],
+          alpha=alpha,
+          zorder=-j,
+      )
+      self._patches.append(self.ax.add_patch(points))
+
+    for i in range(self._k):
+      for j, val in enumerate(ot.x_s[i]):
+        self._points.append(
+            self.ax.scatter(
+                val[0],
+                val[1],
+                s=200 * ot.a_s[i][j] * len(ot.a_s[i]),
+                marker=markers[i],
+                c="black",
+                linewidth=0.0,
+                edgecolor=None,
+                label=str(i)
+            )
+        )
+
+    if self._title is not None:
+      self.ax.set_title(self._title)
+
+    return self._points + self._patches
+
+  def update(
+      self,
+      ot: mmsinkhorn.MMSinkhornOutput,
+      title: Optional[str] = None
+  ) -> List["plt.Artist"]:
+    """Update a plot with a transport instance."""
+    n_s = [len(ot.x_s[i]) for i in range(len(ot.x_s))]
+    assert self._n < jnp.prod(jnp.array(n_s)), \
+      "Intended number of tuples too large."
+    # Extract top_k largest entries in the tensor, and their indices.
+    val, idx = jax.lax.top_k(ot.tensor.ravel(), self._top_k)
+    indices = jnp.unravel_index(idx, n_s)
+
+    alphas = np.linspace(0.6, 0.2, self._top_k - self._n)
+    for j in range(self._top_k):
+      points = [ot.x_s[i][indices[i][j], :] for i in range(self._k)]
+      # reorder to ensure polygons have maximal area
+      points = [points[i] for i in ccworder(jnp.array(points))]
+      alpha = 0.6 if j < self._n else alphas[j - self._n]
+      self._patches[j].set_xy(points)
+      self._patches[j].set_color(self._cmap[j > self._n])
+      self._patches[j].set_alpha(alpha)
+
+    for i in range(self._k):
+      for j, val in enumerate(ot.x_s[i]):
+        idx = np.ravel_multi_index((i, j), (self._k, self._n))
+        self._points[idx].set_offsets(val)
+
+    if title is not None:
+      self.ax.set_title(title)
+
+    self.ax.set_ylim(-2.5e-2, 1 + 2.5e-2)
+    self.ax.set_xlim(-2.5e-2, 1 + 2.5e-2)
+    return self._points + self._patches
+
+  def animate(
+      self,
+      transports: Sequence[Transport],
+      titles: Optional[Sequence[str]] = None,
+      frame_rate: float = 10.0,
+      **kwargs
+  ) -> "animation.FuncAnimation":
+    """Make an animation from several transports."""
+    self._k = len(transports[0].tensor.shape)
+    self._n = transports[0].tensor.shape[0]
+
+    self._top_k = kwargs.pop("top_k", self._n)
```

Comment:
Would not save this, instead pass it to `__call__/update`.

Reply:
Also, why `top_k` should be by default `self._n`?

Reply:
because this is the minimal set of mapping, for each point single polygon